_posts/2020-10-19-empty-byte-array-copy-bug.md

```diff
@@ -0,0 +1,53 @@
+---
+layout: post
+published: true
+title:  "Solidity Empty Byte Array Copy Bug"
+date:   2020-10-19
+author: Solidity Team
+category: Security Alerts
+---
+
+On October 14, 2020, a bug in the Solidity code generator was reported by
+John Toman of the Certora development team. The bug is fixed with [version 0.7.4](https://github.com/ethereum/solidity/releases/tag/v0.7.4)
+released on October 19, 2020. **The bug is present in all prior versions of Solidity.**
+
+We assigned the bug a severity level of "medium".
+
+## Summary
+
+This bug can cause newly created elements of ``bytes`` or ``string`` arrays in storage
+to be initialized by a non-zero value. For this to happen, the following three
+actions have to take place:
+
+ - You copy an **emtpy** ``bytes`` or ``string`` value from ``memory`` or ``calldata`` to storage.
+ - You extend the storage value by modifying ``.length`` or using ``.push()`` (not ``.push(c)``).
+ - You read the newly created byte array element without writing to it first.
+
+Note that the last step con also be done implicitly by reading the whole byte array
+or passing it on somewhere.
```

Comment:
```suggestion
or passing it somewhere.
```

Reply:
I's say either ``passing it on to somewhere``or just ``passing it somewhere``.

Reply:
I'm fine with either of the two.